### PR TITLE
Add option to disable retries in HTTPClient

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes failing EtE tests

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [x] Do all end-to-end tests pass?
* [ ] ~Any other testing performed?~
    > ~Tested by {Running the Monkey locally with relevant config/running Island/...}~
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/infection_monkey/island_api_client/http_island_api_client.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def connect(
     ):
         try:
             self._http_client.server_url = f"https://{island_server}/api/"
-            self._http_client.get(params={"action": "is-up"})
+            self._http_client.get(params={"action": "is-up"}, disable_retries=True)
```
Comment (VakarisZ) [inline monkey/infection_monkey/island_api_client/http_island_api_client.py:65]:
```
requests.get("https://{island_server}/api/", params={"action": "is-up"})
```

File: monkey/infection_monkey/island_api_client/http_island_api_client.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def connect(
...
         except Exception as err:
             self._http_client.server_url = None
             raise err
```
Comment (VakarisZ) [inline monkey/infection_monkey/island_api_client/http_island_api_client.py:68]:
```suggestion
            raise err
        self._http_client.server_url = f"https://{island_server}/api/"
```

File: monkey/infection_monkey/island_api_client/http_island_api_client.py (hunk 1 above)
Comment (VakarisZ) [inline monkey/infection_monkey/island_api_client/http_island_api_client.py:65]:
```suggestion
            requests.get("https://{island_server}/api/", params={"action": "is-up"})
```

File: monkey/infection_monkey/island_api_client/http_island_api_client.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def connect(
     ):
         try:
             self._http_client.server_url = f"https://{island_server}/api/"
```
Comment (VakarisZ) [inline monkey/infection_monkey/island_api_client/http_island_api_client.py:64]:
```suggestion

```

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Being able to provide a custom quantity of retries for a specific request of `HTTPClient` is a good improvement, but I think it could be done in a better way. That being said, we don't need to modify the `HTTPClient` at all to fix the issue at hand, so maybe let's not undertake it for now

**Review by VakarisZ — COMMENTED** [review]
Also, consider writing a unit test to make sure that `connect` method only sends a single request.

File: monkey/infection_monkey/island_api_client/http_island_api_client.py (hunk 1 above)
Comment (shreyamalviya) [reply monkey/infection_monkey/island_api_client/http_island_api_client.py:65]: `HTTPClient` is the only place in the Agent which makes requests. I don't think we want to break that boundary.

**Comment by shreyamalviya** [comment]
Superseded by #3130

**Pull request closed without merging**